src/celeritas/global/alongstep/AlongStepRZMapFieldMscAction.cu

```diff
@@ -43,11 +46,21 @@ void AlongStepRZMapFieldMscAction::execute(CoreParams const& params,
             *this, "propagate-rzmap");
         launch_kernel(params, state, *this, execute_thread);
     }
-    detail::launch_apply_msc(
-        *this, msc_->ref<MemSpace::native>(), params, state);
+    if (this->has_msc())
+    {
+        detail::launch_apply_msc(
+            *this, msc_->ref<MemSpace::native>(), params, state);
+    }
     detail::launch_update_time(*this, params, state);
-    detail::launch_apply_eloss(
-        *this, fluct_->ref<MemSpace::native>(), params, state);
+    if (this->has_fluct())
+    {
+        detail::launch_apply_eloss(
+            *this, fluct_->ref<MemSpace::native>(), params, state);
+    }
+    else
+    {
+        detail::launch_apply_eloss(*this, params, state);
+    }
```

Comment:
I thought briefly about what it would take to make a general templated action (CUDA and C++) that would prevent this logic from being repeated across actions... we can revisit that later.